### PR TITLE
[FAD-102] pool progress info

### PR DESCRIPTION
**Pull request by luansievers**
### Description

> Add Pool progress info component
> Refactor on folder structure of storybook

https://weavik.atlassian.net/browse/FAD-102

https://user-images.githubusercontent.com/14227409/193834443-0f28e653-39c7-4496-b94f-a2e68c60b2dd.mov

### CODE REVIEW

File: packages/client3/components/pool/pool-information/pool-information.tsx [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+  const isUserVerified = true;
+  const canUserParticipate = true;
+
+  /** TODO: REMOVE <- JUST TO TEST */
+
+  const rhfMethods = useForm<FormFields>({
+    mode: "onSubmit",
+    shouldFocusError: true,
+  });
+  // const { control, register, formState } = rhfMethods;
```
Comment (sachasmart-weavik) [inline packages/client3/components/pool/pool-information/pool-information.tsx:94]: necessary?

File: packages/client3/components/pool/pool-information/pool-information.tsx [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+  const { control, watch, register, setValue } = rhfMethods;
+  const [availableBalance, setAvailableBalance] = useState(BigNumber.from(0));
+
+  const validateMaximumAmount = async (value: string) => {
+    if (!account || !usdcContract) {
+      return;
+    }
+    const valueAsUsdc = utils.parseUnits(value, USDC_DECIMALS);
+    console.log(valueAsUsdc);
```
Comment (sachasmart-weavik) [inline packages/client3/components/pool/pool-information/pool-information.tsx:103]: Needed?

File: packages/client3/components/pool/pool-information/pool-information.tsx [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+  const onSubmit = async (data: FormFields) => {
+    debugger;
+    console.log(data);
+    setIsLoading(true);
+    if (!usdcContract || !provider || !account) {
+      throw new Error("Wallet not connected properly");
+    }
+    await signAgreement(account, data.backerName, tranchedPoolAddress);
+
+    // Ensures the user doesn't leave any dust behind when they choose to supply max
```
Comment (sachasmart-weavik) [inline packages/client3/components/pool/pool-information/pool-information.tsx:129]: lol what is dust?

File: packages/client3/components/pool/pool-information/pool-information.tsx [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+    ) {
+      return "Amount exceeds USDC balance";
+    }
+  };
+  //
+  const [isLoading, setIsLoading] = useState(false);
+
+  const onSubmit = async (data: FormFields) => {
+    debugger;
+    console.log(data);
```
Comment (sachasmart-weavik) [inline packages/client3/components/pool/pool-information/pool-information.tsx:122]: necessary?

File: packages/client3/components/pool/pool-information/pool-information.tsx [hunk 5]
```diff
@@ -0,0 +1,333 @@
...
+    }
+    if (
+      valueAsUsdc.gt(availableBalance) &&
+      !usdcWithinEpsilon(valueAsUsdc, availableBalance)
+    ) {
+      return "Amount exceeds USDC balance";
+    }
+  };
+  //
+  const [isLoading, setIsLoading] = useState(false);
```
Comment (sachasmart-weavik) [inline packages/client3/components/pool/pool-information/pool-information.tsx:118]: Can you put this up with all the other hooks? Maybe `line 72`

File: packages/client3/components/pool/pool-information/pool-information.tsx [hunk 6]
```diff
@@ -0,0 +1,333 @@
...
+  const usdcContract = useContract("USDC");
+  const user = useUser() as SupplyPanelUserFieldsFragment;
+
+  // const isUserVerified =
+  //   user?.isGoListed ||
+  //   user?.isUsEntity ||
+  //   user?.isNonUsEntity ||
+  //   user?.isUsAccreditedIndividual ||
+  //   user?.isUsNonAccreditedIndividual ||
+  //   user?.isNonUsIndividual;
```
Comment (sachasmart-weavik) [inline packages/client3/components/pool/pool-information/pool-information.tsx:79]: Should be able to use this without the test.

File: packages/client3/components/pool/pool-information/pool-information.tsx [hunk 7]
```diff
@@ -0,0 +1,333 @@
...
+
+  // const canUserParticipate = user
+  //   ? canUserParticipateInPool(allowedUidTypes, user)
+  //   : false;
+  /** TODO: REMOVE -> JUST TO TEST */
+  const isUserVerified = true;
```
Comment (sachasmart-weavik) [inline packages/client3/components/pool/pool-information/pool-information.tsx:85]: if you've minted a UID in test mode then you'll be `isGoListed`

File: packages/client3/pages/api/accounts/accounts.json [hunk 8]
```diff
@@ -1 +1 @@
-{"accounts":[{"wallet":"0x108Cc3833CD49333A7908e4bB52f4CF8f4090425","role":"Artist"}]}
\ No newline at end of file
+{"accounts":[{"wallet":"0x108Cc3833CD49333A7908e4bB52f4CF8f4090425","role":"Artist"},{"wallet":"0x062765d960DEa3e15cdeB8313ECE1b556FE23FCf","role":"Artist"}]}
```
Comment (sachasmart-weavik) [inline packages/client3/pages/api/accounts/accounts.json]: Whoops, we can take these out. Maybe add to the .gitignore?

File: packages/protocol/.openzeppelin/unknown-31337.json [hunk 9]
```diff
@@ -103,12 +103,12 @@
...
     {
       "address": "0x0Cf17D5DcDA9cF25889cEc9ae5610B0FB9725F65",
-      "txHash": "0x3567b44a17242693870e5d99ea88768b6ed9e4a83960947aa1d4cf8702746c99",
+      "txHash": "0x4856c884c557655364094dd13950a811e7b1e3ded7a37aafb5a3a5e2a7ab260b",
       "kind": "transparent"
     },
     {
       "address": "0xC76A71C4492c11bbaDC841342C4Cb470b5d12193",
-      "txHash": "0x4c4ea01d16f23ac80819ca45188feccc1eda120ed154877c5e2128ec4d821b7b",
+      "txHash": "0x3435cbcc7ab53e29e79ff1c1293260adfa495a66ee6c2750e220ecfe9f5a4cc2",
```
Comment (sachasmart-weavik) [inline packages/protocol/.openzeppelin/unknown-31337.json:111]: No idea what these are either, but I try not to commit them incase it messes up you and Vineeth's stuff.

**Review by sachasmart-weavik — CHANGES_REQUESTED** [review]
Some suggestions. Let me know if you have questions.

File: packages/client3/components/pool/pool-information/pool-information.tsx [hunk 10]
```diff
@@ -0,0 +1,333 @@
...
+    }
+    if (valueAsUsdc.lte(BigNumber.from(0))) {
+      return "Must deposit more than 0";
+    }
+    if (
+      valueAsUsdc.gt(availableBalance) &&
+      !usdcWithinEpsilon(valueAsUsdc, availableBalance)
+    ) {
+      return "Amount exceeds USDC balance";
+    }
```
Comment (v1n33th) [inline packages/client3/components/pool/pool-information/pool-information.tsx:115]: These validations could be extracted out for reusability I guess?

File: packages/client3/components/pool/pool-information/pool-information.tsx [hunk 11]
```diff
@@ -0,0 +1,333 @@
...
+  agreement,
+  tranchedPoolAddress,
+  allowedUidTypes,
+  remainingJuniorCapacity,
+}: Props) {
+  const apolloClient = useApolloClient();
+  const { account, provider } = useWallet();
+  const tranchedPoolContract = useContract("TranchedPool", tranchedPoolAddress);
+  const usdcContract = useContract("USDC");
+  const user = useUser() as SupplyPanelUserFieldsFragment;
```
Comment (v1n33th) [inline packages/client3/components/pool/pool-information/pool-information.tsx:71]: Would suggest adding this type to useUser hook than explicit type casting..

**Review by v1n33th — APPROVED** [review]
Awesome...Some minor suggestions

File: packages/client3/pages/api/accounts/accounts.json (hunk 8 above)
Comment (luansievers) [reply packages/client3/pages/api/accounts/accounts.json]: I think we gonna need to keep this information just like the metadata from pools

File: packages/client3/components/pool/pool-information/pool-information.tsx (hunk 10 above)
Comment (luansievers) [reply packages/client3/components/pool/pool-information/pool-information.tsx]: fixed

**Review by sachasmart-weavik — APPROVED** [review]

**Pull request merged**